rust_snuba/benches/processors.rs

```diff
@@ -176,199 +102,94 @@ fn create_stream_processor(
     StreamProcessor::new(consumer, consumer_state)
 }
 
-fn functions_payload() -> KafkaPayload {
-    let data = r#"{
-        "project_id": 22,
-        "profile_id": "7329158c39964fbb9ec57c20cf4a2bb8",
-        "transaction_name": "vroom-vroom",
-        "timestamp": 1694447692,
-        "functions": [
-            {
-                "fingerprint": 123,
-                "function": "foo",
-                "package": "bar",
-                "in_app": true,
-                "self_times_ns": [1, 2, 3]
-            },
-            {
-                "fingerprint": 456,
-                "function": "baz",
-                "package": "qux",
-                "in_app": false,
-                "self_times_ns": [4, 5, 6]
-            }
-        ],
-        "platform": "python",
-        "environment": "prod",
-        "release": "foo@1.0.0",
-        "dist": "1",
-        "transaction_op": "http.server",
-        "transaction_status": "ok",
-        "http_method": "GET",
-        "browser_name": "Chrome",
-        "device_class": 2,
-        "retention_days": 30
-    }"#;
-    KafkaPayload::new(None, None, Some(data.as_bytes().to_vec()))
+fn run_bench(
+    bencher: &mut BenchmarkGroup<WallTime>,
+    concurrency: usize,
+    make_payload: fn() -> KafkaPayload,
+    processor: &str,
+    schema: &str,
+) {
+    bencher.bench_with_input(
+        BenchmarkId::from_parameter(concurrency),
+        &concurrency,
+        |b, &s| {
+            b.iter(|| {
+                let mut processor = black_box(create_stream_processor(
+                    s,
+                    processor,
+                    schema,
+                    make_payload,
+                    black_box(MSG_COUNT),
+                ));
+                loop {
+                    let res = processor.run_once();
+                    if matches!(res, Err(RunError::Poll(ConsumerError::EndOfPartition))) {
+                        // FIXME: this pretty much means that we *polled* the partition to the end,
+                        // it does not mean that we actually *committed* everything
+                        // (aka we finished actually processing everything).
+                        break;
+                    }
```

Comment:
I believe you can also properly `close` the processor now, as the underlying deadlock was fixed.

Reply:
Should I just call `processor.shutdown()` on EndOfPartition error, and break?